Makefile

```diff
@@ -546,7 +546,7 @@ generate-doctoc:
 	TRACE=$(TRACE) ./hack/generate-doctoc.sh
 
 .PHONY: generate-e2e-templates
-generate-e2e-templates: $(KUSTOMIZE) $(addprefix generate-e2e-templates-, v0.3 v0.4 v1.0 v1.5 v1.6 main) ## Generate cluster templates for all versions
+generate-e2e-templates: $(KUSTOMIZE) $(addprefix generate-e2e-templates-, v0.3 v0.4 v1.0 v1.6 v1.7 main) ## Generate cluster templates for all versions
```

Comment:
@sbueringer @chrischdi should we drop E2E tests for v0.3? (might be on a separated PR, just noticed here)

Reply:
We still use them for clusterctl upgrade `v0.3 => v1.5 => current` tests as of now.

v1.5 must also stay here as long as we have tests like `When testing clusterctl upgrades (v1.5=>current)` in `clusterctl_upgrades_test.go`, but let's see what is all getting removed in this PR.

Reply:
I would like to keep them. They were added to make sure we don't hit https://github.com/kubernetes-sigs/cluster-api/issues/10051 again. I don't see how we can otherwise we be safe against a regression.